src/hip_memory.cpp

```diff
@@ -1969,14 +1970,34 @@ hipError_t hipMemGetInfo(size_t* free, size_t* total) {
         }
 
         if (free) {
-            // TODO - replace with kernel-level for reporting free memory:
-            size_t deviceMemSize, hostMemSize, userMemSize;
-            hc::am_memtracker_sizeinfo(device->_acc, &deviceMemSize, &hostMemSize, &userMemSize);
-
-            *free = device->_props.totalGlobalMem - deviceMemSize;
-
-            // Deduct the amount of memory from the free memory reported from the system
-            if (HIP_HIDDEN_FREE_MEM) *free -= (size_t)HIP_HIDDEN_FREE_MEM * 1024 * 1024;
+            hsa_agent_t* agent = static_cast<hsa_agent_t*>(device->_acc.get_hsa_agent());
+            if (!agent){
+		return ihipLogStatus(hipErrorInvalidResourceHandle);
+            } else {
+		uint32_t internal_node_id;
+		hsa_status_t err = hsa_agent_get_info(*agent, (hsa_agent_info_t) HSA_AMD_AGENT_INFO_DRIVER_NODE_ID, &internal_node_id);
```

Comment:
We don't need to query for nodeID everytime. So a better idea would be to store this information in ihipDevice_t and initialize the value in https://github.com/ROCm-Developer-Tools/HIP/blob/master/src/hip_hcc.cpp#L504.